hy/cmdline.py

```diff
@@ -366,26 +367,30 @@ def hy2py_main():
     import platform
     module_name = "<STDIN>"
 
-    options = dict(prog="hy2py", usage="%(prog)s [options] FILE",
+    options = dict(prog="hy2py", usage="%(prog)s [options] [FILE]",
                    formatter_class=argparse.RawDescriptionHelpFormatter)
     parser = argparse.ArgumentParser(**options)
+    parser.add_argument("FILE", type=str, nargs='?',
+                        help="Input Hy code (use STDIN if \"-\" or "
+                             "not provided)")
     parser.add_argument("--with-source", "-s", action="store_true",
                         help="Show the parsed source structure")
     parser.add_argument("--with-ast", "-a", action="store_true",
                         help="Show the generated AST")
     parser.add_argument("--without-python", "-np", action="store_true",
                         help=("Do not show the Python code generated "
                               "from the AST"))
-    parser.add_argument('args', nargs=argparse.REMAINDER,
-                        help=argparse.SUPPRESS)
 
     options = parser.parse_args(sys.argv[1:])
 
-    if not options.args:
-        parser.exit(1, parser.format_help())
+    stdin_text = None
+    if options.FILE is None or options.FILE == '-':
+        stdin_text = sys.stdin.read()
 
     if options.with_source:
-        hst = pretty_error(import_file_to_hst, options.args[0])
+        hst = (pretty_error(import_file_to_hst, options.FILE)
```

Comment:
But `pretty_error_partial` would have only one use, on the line starting `hst = ...`, so what's the point? You realize that `_ast = ...` uses another two functions, namely `import_file_to_ast` and `import_buffer_to_ast`, not `import_file_to_hst` and `import_buffer_to_hst`, right? Or maybe I'm just having a brain fart. I'd recommend just pushing the edit you want to make instead of trying to explain it to me.